### PR TITLE
medor:  7 - Dates d'abonnement

### PR DESCRIPTION
**Pull request by robinkeunen**
[ref](https://gestion.coopiteasy.be/web#id=2298&view_type=form&model=project.task&action=479)

### CODE REVIEW

File: product_subscription/models/subscription.py [hunk 1]
```diff
@@ -288,6 +303,31 @@ def create(self, vals):
...
 
+    @api.multi
+    @api.depends('subscribed_on', 'template')
+    def _compute_date_end(self):
+        for subscription in self:
+            if not subscription.subscribed_on:
+                return False
+
+            elif subscription.template.time_span_unit == 'year':
+                subscription.end_date = add_years(
```
Comment (houssine78) [inline product_subscription/models/subscription.py:316]: be careful when computing the subscription end date on the subscription_on field. This field give the first subscription date. It doesn't give the start date of the ongoing subscription which could have been renewed. It that case the end date will be wrong.

I propose to add a start date which will be the same than the subscribed_on for a new subscription and the effective payment date for a renew.

**Review by houssine78 — CHANGES_REQUESTED** [review]
make change accordingly to my comment.

File: product_subscription/models/subscription.py (hunk 1 above)
Comment (robinkeunen) [reply product_subscription/models/subscription.py:316]: Good comment but the subscriptions are not renewed anymore. Every new request is a new subscription. It is therefore always the subscription date.

Comment (houssine78) [reply product_subscription/models/subscription.py:316]: This is madness !

https://tenor.com/view/gerald-butler-madness-mad-angry-insane-gif-7844903

**Review by houssine78 — APPROVED** [review]

**Pull request merged**